t/redirects.rb

```diff
@@ -0,0 +1,43 @@
+require 'test_helper'
+
+describe 'redirecting links' do
+  around { |test| VCR.use_cassette(titles.first, &test) }
+  subject { WikiData.ids_from_pages('en', titles) }
+
+  describe 'redirect and direct link' do
+    let(:titles) { ['The Meaning of Liff', 'The Deeper Meaning of Liff'] }
+
+    it 'pairs both redirecting link and direct link with wikidata ID' do
+      subject.must_equal('The Meaning of Liff' => 'Q875382', 'The Deeper Meaning of Liff' => 'Q875382')
+    end
+  end
+
+  describe 'only redirect links' do
+    let(:titles) { ['Marvin the Paranoid Android', 'The Deeper Meaning of Liff'] }
+
+    it 'returns wikidata ids paired with redirecting links and direct links' do
+      subject.must_equal('Marvin (character)'          => 'Q264685',
+                         'The Meaning of Liff'         => 'Q875382',
+                         'The Deeper Meaning of Liff'  => 'Q875382',
+                         'Marvin the Paranoid Android' => 'Q264685')
+    end
+  end
+
+  describe 'multiple links redirecting to the same article' do
+    let(:titles) { ['Wikkit Gate', 'Infinite Improbability Drive'] }
+
+    it 'returns wikidata ids paired with redirecting links and direct links' do
+      subject.must_equal("Technology in The Hitchhiker's Guide to the Galaxy" => 'Q259299',
+                         'Infinite Improbability Drive'                       => 'Q259299',
+                         'Wikkit Gate'                                        => 'Q259299')
+    end
+  end
+
+  describe 'page without redirects' do
```

Comment:
The ordering here makes this quite hard to follow. In general it's best to think of the code being read linearly — i.e. you should start with the simplest cases, and built up to the more complex ones.